tools/dptools/src/dptools/bandout.py

```diff
@@ -62,35 +65,35 @@ def fromfile(cls, fobj):
         ispins = set()
         kweights = []
         eigvalarrays = []
-        match = _PAT_BLOCK.search(txt)
-        while match:
+        ndatacols = 0
+        for match in  _PAT_BLOCK.finditer(txt):
             ispin = match.group("ispin")
-            if ispin:
-                ispins.add(int(ispin))
-            else:
-                ispins.add(1)
+            ispins.add(int(ispin) if ispin is not None else 1)
             kweight = match.group("kweight")
-            if kweight:
-                kweights.append(float(kweight))
-            else:
-                kweights.append(1.0)
+            kweights.append(float(kweight) if kweight is not None else 1.0)
+
             vals = match.group("vals")
             tmp = np.array(vals.split(), dtype=float)
-
-            # Only keep last two data columns
-            # (optional first column, if present, contains sequential numbering)
             nrows = vals.strip().count('\n') + 1
             ncols = len(tmp) // nrows
-            tmp.shape = (nrows, ncols)
-            tmp = tmp[:, ncols - 2 : ncols]
 
+            # Optional first row might contain sequential numbering.
```

Comment:
First row or first column? I am not sure if I understand correctly the following block. For me it makes sense if the first *column* would be sequential numbers. If is that the case and I understood correctly, the script just skip that first column, right? My second question is, when do we have those sequential numbering? Is a matter of previous versions output?

Reply:
That's a typo, I meant first column of course. Thx, corrected.

As for the sequential numbering: Eigenlevels in `band.out` are numbered for easier interpretation by humans. In the `ProjectedStates` output files, however, this enumeration is missing. Therefore, the script must be able to handle both cases.

Reply:
You are welcome. Understood. Thanks, Bálint!